### ISSUE TITLE
Support moving folders through the quickMove panel

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#75]
At the moment QuickFolders supports moving folders via QuickFolders tabs, by holding the SHIFT key. If would be very helpful if folders could also be dragged onto the quickMove button so that they can be moved to any desired target location:

![image](https://user-images.githubusercontent.com/6314674/99902411-89f26900-2cb5-11eb-9083-5271db00d713.png)

**Comment by RealRaven2000** [comment]
I had a request of providing a (dedicated?) keyboard shortcut for moving folders, but I reckon - since free shortcut keys are very hard to find - we could keep using the same (Shift+M) shortcut provided the mouse / key focus is on the folder tree.

In that case, quickMove should give a warning message box, as all current users could assume it was there to move highlighted mail only. If we displayed this message (only if action was triggered by the shortcut, not on mouse drag):

```
This will move the selected folders - are you sure? 
[Ok] [Cancel]
```
you could just hit Enter and we would prevent some bad accidents happening for those.

**Comment by ThomasKjaergaard** [comment]
Reusing the SHIFT+M shortcut will be fine for me.

The moving by dragging a folder to the move icon does not work for me.
I did hold SHIFT down while trying it.

**Comment by RealRaven2000** [comment]
> Reusing the SHIFT+M shortcut will be fine for me.
> 
> The moving by dragging a folder to the move icon does not work for me.

That's expected, as this issue #75 is about implementing it. :)

At the moment this is handled by the same code that handles dropping a folder to a QuickFolders tab, so you should be seeing this warning:
![image](https://user-images.githubusercontent.com/6314674/104025054-f8cf4700-51bb-11eb-8eab-8c5bc9e6a4bd.png)

I need to add a bunch of special code for having the quickMove button as drop target.

**Comment by ThomasKjaergaard** [comment]
> However - if you hold down SHIFT and drop to a QuickFolders tab, moving folders already works!

That works. As I move most folders to the same folder, that is a big improvement to the situation before.
that was dragging a folder in the folder tree, and waiting for the scrolling to come to the right location. Which can take long tie with many folders.

**Comment by RealRaven2000** [comment]
> > However - if you hold down SHIFT and drop to a QuickFolders tab, moving folders already works!
> 
> That works. As I move most folders to the same folder, that is a big improvement to the situation before.
> that was dragging a folder in the folder tree, and waiting for the scrolling to come to the right location. Which can take long tie with many folders.

It also supports moving multiple selected folders in one operation - that's something that you don't get in the tree (and it's kind of dangerous). The less you use the tree, the least painful your life will become. Mind got a whole lot better after deciding not to use the tree for 2 things:
* folder navigation
* moving mail

This can all be done with QuickFolder Tabs and the quickMove / quickJump functions + navigation buttons shown on the current folder bar / shortcuts.

**Comment by RealRaven2000** [comment]
Here is a initial test version:

[QuickFolders-wx-5.2pre16.zip](https://github.com/RealRaven2000/QuickFolders/files/5792616/QuickFolders-wx-5.2pre16.zip)

it's not well tested for if you drag folders and then mails before selecting a target, I guess it should be able to do both in one operation (a mix of folders and mails), that might just lead to some incomplete moving at the moment. At first I had a confirmation message on the drop event but since there is already on when you select the target folder, I thought this was overkill. So you will see exactly one message when selecting the target:
![image](https://user-images.githubusercontent.com/6314674/104123915-b46fb280-5345-11eb-8d70-39866bc22692.png)

**Comment by RealRaven2000** [comment]
In my tests moving multiple folders on an Imap account proves somewhat unreliable. This may be due to the fact that I break up the moving into multiple calls to nsIMsgCopyService.CopyFolders(..) partly because some folders can only be copied and this is something I find out per folder before that call. Thinking of moving them together into one call and hope for the best?

**Comment by RealRaven2000** [comment]
Here is an improved version, which supports moving multiple folders via quickMove - there were complications of you highlight child folders which may lead to an incomplete moving of a selection like this:

![image](https://user-images.githubusercontent.com/6314674/105219350-464d9b80-5b4e-11eb-82c7-4d3de8dbc510.png)

In the example, QuickFolders would attempt to move "test3" (A childfolder of "atn") after its parent was moved and so lead to an error - potentially then also not moving the following folders "filtaquilla" and "quickfilters". This version will remove all "child folders" from the array of moved folders and so run much more stable:

[QuickFolders-wx-5.2pre23.zip](https://github.com/RealRaven2000/QuickFolders/files/5844530/QuickFolders-wx-5.2pre23.zip)

There is also a back-ported version for Thunderbird 68 users:
[QuickFolders-wx-4.19pre10.zip](https://github.com/RealRaven2000/QuickFolders/files/5844547/QuickFolders-wx-4.19pre10.zip)

Note: when you move the four folders as of the example the confirmation message will ask about the 3 "legal" folders but it will move everything as expected.

**Comment by RealRaven2000** [comment]
Implemented in Releases 5.2 and 4.19.1 (published on 22/01/2021)

**Comment by ThomasKjaergaard** [comment]
I can confirm it works in version 5.2.
Both dragging and with SHIFT+m.

Thank you.
You made my mail sorting easier.
